OpenRA.Mods.Common/Widgets/CheckboxWidget.cs

```diff
@@ -76,15 +82,21 @@ public override void Draw()
 				font.DrawText(text, position,
 					disabled ? colordisabled : color);
 
-			if (IsChecked() || (Depressed && HasPressedState && !disabled))
+			if (getCheckmarkImageCache == null)
 			{
-				var checkType = GetCheckType();
-				if (HasPressedState && (Depressed || disabled))
-					checkType += "-disabled";
-
-				var offset = new float2(rect.Left + CheckOffset, rect.Top + CheckOffset);
-				WidgetUtils.DrawSprite(ChromeProvider.GetImage("checkbox-bits", checkType), offset);
+				getCheckmarkImageCache = new CachedTransform<(string, bool, bool), CachedTransform<(bool, bool, bool, bool), Sprite>>(
+					((string CheckType, bool Highlighted, bool Checked) args) =>
+					{
+						var variantImageCollection = args.Highlighted ? "checkmark-highlighted-" : "checkmark-" + args.CheckType;
+						var variantBaseName = args.Checked ? "checked" : "unchecked";
+						return WidgetUtils.GetCachedStatefulImage(variantImageCollection, variantBaseName);
+					});
 			}
+
+			var checkedImage = getCheckmarkImageCache
+				.Update((GetCheckmarkType(), IsHighlighted(), IsChecked()))
+				.Update((disabled, Depressed, hover, false));
+			WidgetUtils.DrawSprite(checkedImage, new float2(rect.Left + CheckOffset.X, rect.Top + CheckOffset.Y));
```

Comment:
I suggest getting rid of `CheckOffset` and using this logic to automatically center the glyph in the checkbox:
```csharp
WidgetUtils.DrawSprite(checkedImage, new float2(check.Right - (int)((check.Height + checkedImage.Size.X) / 2), check.Top + (int)((check.Height - checkedImage.Size.Y) / 2)));
```

This pattern is used in a couple of other widgets and I think is worth pulling out to `WidgetUtils`. But that can be done in a follow up unless we want to scope creep it here.

Reply:
Implemented

I'm going to avoid scope creep here, I want these pr's merged afterall